src/db.c

```diff
@@ -1826,6 +1826,25 @@ keyStatus expireIfNeededWithDictIndex(serverDb *db, robj *key, int flags, int di
     if (server.primary_host != NULL) {
         if (server.current_client && (server.current_client->flag.primary)) return KEY_VALID;
         if (!(flags & EXPIRE_FORCE_DELETE_EXPIRED)) return KEY_EXPIRED;
+    } else if (server.import_mode) {
+        /* If we are running in the import mode on a primary, instead of
+         * evicting the expired key from the database, we return ASAP:
+         * the key expiration is controlled by the import source that will
+         * send us synthesized DEL operations for expired keys. The
+         * exception is when write operations are performed on this server
+         * because it's a primary.
+         *
+         * Notice: other clients, apart from the import source, should not access
+         * the data imported by import source.
```

Comment:
Shouldn't this be enforced by server instead of providing a warning in the code comment/config file? The client's not marked with import_source should receive `LOADING` error if I'm understanding the use case correctly.

Reply:
We have discussed this issue before. In real-world scenarios, there are not only cases where data is migrated to an empty instance, but also cases where data is migrated to an instance that already contains data (requiring the user to ensure that the data being migrated is not disrupted). We hope to continue providing normal services in this scenario.

Reply:
Yes, the feature can cause some inconsistencies if misused, but we decided that it's better to let the user have the power than to limit them.

Reply:
Ack.